### ISSUE TITLE
[Java]  Serializing record with "String" member fails with NPE

### ISSUE THREAD

**Issue by LumnitzF** [alipay/fury#1003]
**Describe the bug**
Serializing a record with a String member fails with an NPE

**Expected behavior**
Serialization works as expected

**To Reproduce**
See this test:
```java
import org.junit.jupiter.api.Test;

import io.fury.Fury;

class MyTest {

	@Test
	void foo() {
		Fury fury = Fury.builder().build();
		fury.register(Foo.class);
		byte[] serialized = fury.serialize(new Foo("foo")); // fails
		Object deserialized = fury.deserialize(serialized);
		System.out.println(deserialized);
	}

	private record Foo(String foo) {
	}

}
```

Stacktrace:
```
java.lang.RuntimeException: Create sequential serializer failed, 
class: class MyTest$Foo

	at io.fury.serializer.CodegenSerializer.loadCodegenSerializer(CodegenSerializer.java:46)
	at io.fury.resolver.ClassResolver.lambda$getObjectSerializerClass$2(ClassResolver.java:943)
	at io.fury.builder.JITContext.registerSerializerJITCallback(JITContext.java:132)
	at io.fury.resolver.ClassResolver.getObjectSerializerClass(ClassResolver.java:941)
	at io.fury.resolver.ClassResolver.getSerializerClass(ClassResolver.java:896)
	at io.fury.resolver.ClassResolver.getSerializerClass(ClassResolver.java:804)
	at io.fury.resolver.ClassResolver.createSerializer(ClassResolver.java:1153)
	at io.fury.resolver.ClassResolver.getOrUpdateClassInfo(ClassResolver.java:1095)
	at io.fury.Fury.writeRef(Fury.java:283)
	at io.fury.Fury.serialize(Fury.java:235)
	at io.fury.Fury.serialize(Fury.java:183)
	at MyTest.foo(MyTest.java:11)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.NullPointerException: Cannot read field "f0" because "methodInfo" is null
	at io.fury.builder.CodecBuilder.getRecordFieldValue(CodecBuilder.java:201)
	at io.fury.builder.CodecBuilder.getFieldValue(CodecBuilder.java:151)
	at io.fury.builder.ObjectCodecBuilder.lambda$serializeGroup$9f8a61a4$1(ObjectCodecBuilder.java:186)
	at io.fury.builder.ObjectCodecBuilder.serializeGroup(ObjectCodecBuilder.java:195)
	at io.fury.builder.ObjectCodecBuilder.buildEncodeExpression(ObjectCodecBuilder.java:159)
	at io.fury.builder.BaseObjectCodecBuilder.genCode(BaseObjectCodecBuilder.java:204)
	at io.fury.codegen.CompileUnit.getCode(CompileUnit.java:54)
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:73)
	at io.fury.codegen.JaninoUtils.toBytecode(JaninoUtils.java:65)
	at io.fury.codegen.CodeGenerator.compile(CodeGenerator.java:141)
	at io.fury.builder.CodecUtils.loadOrGenCodecClass(CodecUtils.java:89)
	at io.fury.builder.CodecUtils.loadOrGenObjectCodecClass(CodecUtils.java:40)
	at io.fury.serializer.CodegenSerializer.loadCodegenSerializer(CodegenSerializer.java:43)
	... 14 more
```

**Environment:**
Fury 0.2.0, Windows 10, Eclipse Adoptium 17.0.8.7

**Comment by chaokunyang** [comment]
Thanks for creating this issue, I fixed it in https://github.com/alipay/fury/pull/1004, could you try it out?

**Comment by chaokunyang** [comment]
Hi, we've just released fury 0.2.1: https://github.com/alipay/fury/releases/tag/v0.2.1 .It should have fixed this issue. Could you try it out?
